### PR TITLE
subject database hooks changes to update redis entries

### PR DESCRIPTION
**Pull request by shriramshankar**

### CODE REVIEW

File: cache/sampleStore.js [hunk 1]
```diff
@@ -101,7 +101,7 @@ function removeNullsAndStringifyArrays(obj, arrayFields) {
  * @returns {Object} cleaned up and ready to store in redis.
  */
 function cleanAspect(a) {
-  let retval = a.get();
+  let retval = a.get ? a.get() : a;
```
Comment (iamigo) [inline cache/sampleStore.js]: maybe add a comment here or on line 100 suggesting that "a" can either be a sequelize "instance" object OR just a regular object?

File: cache/sampleStore.js [hunk 2]
```diff
@@ -114,7 +114,7 @@ function cleanAspect(a) {
  * @returns {Object} cleaned up and ready to store in redis.
  */
 function cleanSample(s) {
-  let retval = s.get();
+  let retval = s.get ? s.get() : s;
```
Comment (iamigo) [inline cache/sampleStore.js]: same comment as above

File: package.json [hunk 3]
```diff
@@ -43,6 +43,7 @@
     "test-disablehttp": "DISABLE_HTTP=true mocha -R dot --recursive tests/disableHttp",
     "test-enforced": "USE_ACCESS_TOKEN=true mocha -R dot --recursive tests/enforceToken",
     "test-db": "npm run checkdb && mocha -R dot --recursive tests/db",
+    "test-cache" : "mocha -R dot --recursive tests/cache",
```
Comment (iamigo) [inline package.json]: is this just for convenience during dev?

Comment (shriramshankar) [inline package.json]: yes just for convenience.

**Comment by shriramshankar** [comment]
@iamigo PR updated.

File: cache/sampleStore.js [hunk 4]
```diff
@@ -97,11 +97,12 @@ function removeNullsAndStringifyArrays(obj, arrayFields) {
 /**
  * Remove nulls and stringify arrays.
  *
- * @param {Object} a - The aspect to clean.
+ * @param {Object} a - The aspect to clean. This can be either be a sequelize
+ * object instance or just a a regular object.
```
Comment (iamigo) [inline cache/sampleStore.js]: "... just a **a** regular object"

File: cache/sampleStore.js [hunk 5]
```diff
@@ -110,11 +111,12 @@ function cleanAspect(a) {
 /**
  * Remove the aspect field, remove nulls and stringify arrays.
  *
- * @param {Object} s - The sample to clean.
+ * @param {Object} s - The sample to clean. This can be either be a sequelize
+ * object instance or just a a regular object.
```
Comment (iamigo) [inline cache/sampleStore.js]: "... just a **a** regular object"

File: cache/redisOps.js [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+ */
+function capitalizeFirstLetter(str) {
+  return str.charAt(0).toUpperCase() + str.slice(1);
+} // capitalizeFirstLetter
+
+/**
+ * Creates or updates the hash specified by the name argument, with values
+ * specified by the value argument.
+ * @param  {[type]} objectName - Name of the object.
+ * @param  {[type]} name  - Name used to identify the hash
```
Comment (pallavi2209) [inline cache/redisOps.js]: [Type]

**Comment by shriramshankar** [comment]
@pallavi2209 updated the PR.

**Comment by pallavi2209** [comment]
@shriramshankar Thanks, but I mistakenly just commented on a line. I still need to go through the changes.

File: tests/cache/models/subjects/subjectCRUD.js [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+    .then((ret) => {
+      // since sample store has not been popluated yet. rename should not create
+      // a new entry in sample store.
+      expect(ret.length).to.equal(0);
+      done();
+    })
+    .catch(done);
+  });
+
+  it('when asbolutepath changes, the sampleStore refelct this change', (done) => {
```
Comment (pallavi2209) [inline tests/cache/models/subjects/subjectCRUD.js]: refelct*

**Comment by shriramshankar** [comment]
@iamigo @pallavi2209 PR updated.

**Pull request merged**